From 6a1bc6dcd84c6b887918ebb4bf0ee5b97bf82191 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Thu, 28 Sep 2017 16:47:41 +0300
Subject: [PATCH 2016/2241] fix: dts: a7040_pcac: add sfi comphy handle

Currently 10G port doesn't work on PCAC because comphy was enabled
without setting the SFI comphy handle (the driver reset all
the comphys without configuring it).

In addition, USB comphy handler is set incorrectly to comphy4.

This patch fixes this issue by adding the required comphy
handle for 10G port and removing the incorrect one for USB.

Change-Id: I02ee69c3d5488c1e0a891f4db557dec28c773db8
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/44873
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-pcac.dts | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts b/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
index 6bdacd2..0305630 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-pcac.dts
@@ -132,9 +132,8 @@
 			};
 			usb3h1: usb3@510000 {
 				separated-phys-for-usb2-usb3;
-				phys = <&utmi1>,
-				       <&comphy0 4 COMPHY_USB3H1>;
-				phy-names = "usb2", "usb3";
+				phys = <&utmi1>;
+				phy-names = "usb2";
 				status = "okay";
 			};
 			spi@700680 {
@@ -188,7 +187,9 @@
 
 			gop {
 				emac0: mac0 {
-					phy-mode = "sfi"; /* lane-2 */
+					phy-mode = "sfi"; /* lane-4 */
+					phys = <&comphy0 4 COMPHY_SFI>;
+					phy-names = "comphy";
 				};
 				emac2: mac2 {
 					phy-mode = "rgmii"; /* lane-0 */
-- 
2.7.4

